### ISSUE TITLE
Google Map absorbing clicks to elements higher in a stack

### ISSUE THREAD

**Issue by randomdude583** [marchdev-tk/flutter_google_maps#23]
I have a Stack with a button and a GoogleMap. When I click the Button, the onTap from the button AND the GoogleMap fire.

```
stack(
    children: [
        Positioned.fill(
            child: GoogleMap(
                key: _key,
                markers: Set(),
                onTap: (pos){print("Map Click");},
                onLongPress: (pos){print("Map Long Click");},
                initialPosition: GeoCoord(43.589, -116.405),
            )
        ),
          
        RaisedButton(
            onPressed: (){print("Button click");}, 
                child: Text("Text")
         ),
    ]
)

When moving the cursor over the button, the cursor stays the same, rather than converting back to the default cursor, and when clicking the button, both children fire their onTap events.

How do I stop this behavior from occuring?

Thanks!
```

**Comment by Pizzacorn** [comment]
I have a similar problem too, I have a Container over the Map in a Stack, when I click in the Container elements, the OnTap of the Container and the OnTap of the map fire at the same time.

Also happens with cursors, when I put a Button which is over the Map, the cursor doesn't change, always appears the hand cursor even though you put another one.

**Comment by randomdude583** [comment]
@Pizzacorn,
Yep, that sounds exactly like the behavior that I am seeing.

**Comment by randomdude583** [comment]
Is there any update on this issue?
I would really like to use this in my next roll out.

Thanks,

**Comment by dhs9004** [comment]
I'm having similar issues when I tried to use Marker/onInfoWindowTap on web

**Comment by bobppsureway** [comment]
Same issue. I have a Drawer with ListView, drag listview up and down GoogleMap interactive will fire. Tap ListTiles or any other buttons in Drawer GoogleMap Marker/oninfoWindow onTap or will fire. 
Try to setState on map interactive false doesn't work. 
Try to use transparent Container or GestureDetecor on top layer doesn't work.
Waiting for updates.

**Comment by randomdude583** [comment]
I've been looking into this, and it isnt local to this repository. This repo uses the [google_maps](https://pub.dev/packages/google_maps) plugin to generate the map for web, and it has the same problem. There is also [this ](https://github.com/ditman/plugins/tree/gmaps_platform_interface_bump)repository, which is a early development version of the google_maps_flutter plugin from the flutter.dev team. It has some features that this repo does not have, but also has a lot of missing features, and also has the transparent click issue, as it is built using the same base plugin.

Hope this helps somebody

**Comment by joaquini** [comment]
Wow! I'm with the same issue in web.
I tried wrapping the `GoogleMap` widget into a `IgnorePointer` widget and same happens. 
Somehow, native `GoogleMap` is always consuming view events. In Android it works fine (No action allowed at all when wrapped with an `IgnorePointer`).

**Comment by randomdude583** [comment]
@joaquini, yep, I tried the IgnorePointer as well. Its definitely the JS google map that is consuming all of the touch events. The problem persists in other libraries implementing the google_maps library [here](https://pub.dev/packages/google_maps)

**Comment by davidjan3** [comment]
This issue is still reproducible, and makes this plugin mostly unusable in web applications, since you can't properly put any controls on top of the map.

**Comment by tomrozb** [comment]
> This issue is still reproducible, and makes this plugin mostly unusable in web applications, since you can't properly put any controls on top of the map.

This is a bug in flutter with iframes am will be fixed in framework. We have to wait or look for a workaround

**Comment by joaquini** [comment]
> > This issue is still reproducible, and makes this plugin mostly unusable in web applications, since you can't properly put any controls on top of the map.
> 
> This is a bug in flutter with iframes and will be fixed in framework. We have to wait or look for a good workaround

Do you have a link with official info about the progress on this?

**Comment by tomrozb** [comment]
> > This is a bug in flutter with iframes and will be fixed in framework. We have to wait or look for a good workaround
> 
> Do you have a link with official info about the progress on this?

https://github.com/flutter/flutter/issues/54027

**Comment by yurisasc** [comment]
I'm facing the same issue, is there a possible workaround for this?

**Comment by Scott1803** [comment]
Does anyone have an update on this? I am getting this exact behaviour and cant seem to find any fixes or workarounds

**Comment by DannyJoinedApps** [comment]
Not that anyone is in any doubt but I'll add a plus one. I too have this. Stacks and google maps just don't seem to mix. Not just capturing, or rather listening to all mouse or touch events, but it even stops the cursor changing, e.g. and inkwell on a widget higher in the stack wont cause the cursor to change appropriately. Real shame as screen real estate such a massive issue that hovering buttons over the top of the map is a huge help and while the buttons do also capture a click (any widget requiring a drag gesture is definitely not usable, but single taps are just about okay except...) those widgets don't give a different cursor on mouse-driven devices, so users (especially old ones who don't just try tapping) are convinced a button isn't a button.

**Comment by DannyJoinedApps** [comment]
This:
https://pub.dev/packages/pointer_interceptor
Has got round the problem completely for me.
